src/parser.rs

```diff
@@ -124,3 +126,136 @@ impl Statement {
         }
     }
 }
+
+#[cfg(test)]
+mod statement_prep {
+    use super::*;
+
+    macro_rules! get_statement {
+        ($input: literal) => {
+            Statement::prep(&$input.to_owned());
+        };
+    }
+
+    #[test]
+    fn test_returns_unk_on_unknown_statement() {
+        let statement = get_statement!("UNKNOWN_STATEMENT");
+        assert_eq!(statement, Statement {
+            stype: StatementType::Unk,
+            key: None,
+            value: None
+        });
+    }
+
+    mod get {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_get_without_key() {
+            let statement = get_statement!("GET");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get() {
+            let statement = get_statement!("GET MY_KEY");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Get,
+                key: Some("MY_KEY".to_owned()),
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get_ignoring_next_keys() {
```

Comment:
How about we rename `returns_get_ignoring_next_keys()` as `test_ignoring_multiple_keys_on_get()`. BTW, would this be intended behaviour of a KV database, I am not sure, we could theoretically allow the parser, in the future to allow accessing multiple data-values from the db on a single GET command. Handling failures would be a problem though, given the complexity that we will have to factor in, later.

Reply:
idem